src/Symfony/Bundle/FrameworkBundle/Command/DotenvDebugCommand.php

```diff
@@ -0,0 +1,126 @@
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Dotenv\Dotenv;
+
+/**
+ * A console command to debug current dotenv files with variables and values.
+ *
+ * @author Christopher Hertel <mail@christopher-hertel.de>
+ */
+final class DotenvDebugCommand extends Command
+{
+    protected static $defaultName = 'debug:dotenv';
+    protected static $defaultDescription = 'Lists all dotenv files with variables and values';
+
+    private $kernelEnvironment;
+    private $projectDirectory;
+
+    public function __construct(string $kernelEnvironment, string $projectDirectory)
+    {
+        $this->kernelEnvironment = $kernelEnvironment;
+        $this->projectDirectory = $projectDirectory;
+
+        parent::__construct();
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+        $io->title('Dotenv Variables & Files');
+
+        $envFiles = $this->getEnvFiles();
+        $availableFiles = array_filter($envFiles, function (string $file) {
+            return is_file($this->projectDirectory.\DIRECTORY_SEPARATOR.$file);
+        });
+
+        if (\in_array('.env.local.php', $availableFiles, true)) {
+            $io->warning('Due to existing dump file (.env.local.php) all other dotenv files are skipped.');
+        }
+
+        $io->section('Available Files');
+        $io->listing(array_map(static function (string $envFile) use ($availableFiles) {
+            return \in_array($envFile, $availableFiles, true)
+                ? sprintf('<fg=green>✓</> %s', $envFile)
+                : sprintf('<fg=red>⨯</> %s', $envFile);
+        }, $envFiles));
+        $io->comment('Priority is descending');
+
+        $io->section('Variables with values');
+        $io->table(
+            array_merge(['Variable', 'Value'], $availableFiles),
+            $this->getVariables($availableFiles)
+        );
+
+        $io->comment('Note real values might be different between web and CLI.');
+
+        return 0;
+    }
+
+    private function getVariables(array $envFiles): array
+    {
+        $vars = [];
+        $fileValues = [];
+
+        foreach ($envFiles as $envFile) {
+            $file = $this->projectDirectory.\DIRECTORY_SEPARATOR.$envFile;
+            if (!is_file($file)) {
+                continue;
+            }
+            $values = $this->loadValues($file);
+            $vars = array_merge($vars, array_keys($values));
+            $fileValues[$envFile] = $values;
+        }
+        $vars = array_unique($vars);
+        sort($vars);
+
+        $output = [];
+        foreach ($vars as $var) {
```

Comment:
you should be able to rely on `SYMFONY_DOTENV_VARS`

Reply:
reduces testability, but let's see - for now that's easier :)